File: 49. Group Anagrams/49. Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> anagramGroups;
        
        for (const string& str : strs) {
            string sortedStr = str;
            sort(sortedStr.begin(), sortedStr.end());
            anagramGroups[sortedStr].push_back(str);
        }
        
        vector<vector<string>> result;
        for (const auto& group : anagramGroups) {
            result.push_back(group.second);
        }
        
        return result;
    }
};

File: 49. Group Anagrams/49. Group Anagrams.go
func groupAnagrams(strs []string) [][]string {
    anagramGroups := make(map[string][]string)
    
    for _, str := range strs {
        sortedStr := sortString(str)
        anagramGroups[sortedStr] = append(anagramGroups[sortedStr], str)
    }
    
    result := make([][]string, 0, len(anagramGroups))
    for _, group := range anagramGroups {
        result = append(result, group)
    }
    
    return result
}

func sortString(s string) string {
    chars := []rune(s)
    sort.Slice(chars, func(i, j int) bool {
        return chars[i] < chars[j]
    })
    return string(chars)
}